src/main/java/mateusz/michal/chat/Service/RegistrationService.java

```diff
@@ -83,12 +94,82 @@ private boolean isEmailIncorrect(String email){
         return !pattern.matcher(email).matches();
     }
 
-    public boolean isPasswordIncorrect(String password) {
+    private boolean isPasswordIncorrect(String password) {
         Pattern pattern = Pattern.compile("(?=.*\\d)(?=.*[a-z])(?=.*[A-Z])" +
                 "(?=.*[\\!\\@\\#\\$\\%\\^\\&\\*])(?!=.*\\s).{8,15}");
         return !pattern.matcher(password).matches();
     }
 
+    private List<MyError> validateUserName(String name){
+        List<MyError> errors = new ArrayList<>();
+        if(isNameNull(name)){
+            errors.add(new MyError(400,RegisterFormErrorCode.NAME_NULL,
+                    "parameter name is null"));
+        } else {
+            if(isNameNotPresent(name)){
+                errors.add(new MyError(422,RegisterFormErrorCode.NAME_MISSING,
+                        "parameter name isn't present"));
+            }
+            if(isUserInDatabaseByName(name)){
+                errors.add(new MyError(422,RegisterFormErrorCode.NAME_OCCUPIED,
+                        "In database exist user with that name"));
+            }
+            if(isNameIncorrect(name)){
```

Comment:
Wydaje mi się że lepszym pomysłem jest zamieszczenie walidacji wewnątrz aplikacji(serwera) na początku (przed walidacją na poziomie bazy danych - pozwala uniknąć niepotrzebnego połączenia z bazą)

Reply:
A, nieważne - w końcu za każdym razem zwracasz wszystkie errory - to by miało zastosowanie jakbyś kończył na konkretnym błędzie :) my bad